hypothesis-python/src/hypothesis/core.py

```diff
@@ -154,6 +151,12 @@ def seed(seed):
 
     def accept(test):
         test._hypothesis_internal_use_seed = seed
+        current_settings = getattr(test, "_hypothesis_internal_use_settings", None)
+        if getattr(current_settings, "database", None) is not None:
+            warnings.warn("@seed implies database=None", HypothesisWarning)
```

Comment:
I'm not sure this is a useful warning - it will produce a warning in the default use of seed, and seed isn't really intended for long-term use anyway.